server/src/main/java/io/crate/fdw/JdbcBatchIterator.java

```diff
@@ -150,10 +151,19 @@ public CompletionStage<?> loadNextBatch() throws Exception {
         if (conn == null) {
             conn = DriverManager.getConnection(url, properties);
         }
-        if (statement == null) {
-            statement = conn.prepareStatement(sql);
+        try {
+            if (statement == null) {
+                statement = conn.prepareStatement(sql);
+            }
+            resultSet = statement.executeQuery();
+        } catch (Exception e) {
+            throw new IllegalStateException(
+                String.format(
+                    Locale.ENGLISH,
+                    "The query on the foreign table, '%s' failed due to: %s",
+                    localName.sqlFqn(),
+                    e.getMessage()));
```

Comment:
Or could we introduce ForeignDataWrapperException?

Reply:
I don't think replacing one generic exception type with another adds any value.

We should instead probably map the most common scenarios to specific errors. E.g. if a table is missing, we should raise `RelationUnknown`

This also should have a test case

Reply:
Thank you, I think I can use pg error codes for this.

Btw, shouldn't we support `show create table` on foreign tables? Once the table is created, there is no way of knowing the table structures at the moment.

Reply:
For now I can only think of schema/relation exceptions.

Do you think we can extend `ForeignSchemaUnknown`/`ForeignRelationUnknown` to be more precise? Of the two, `SchemaUnknown` is problematic. `select * from doc.t`, where `doc.t` is mapped to `foreign_schema.foreign_table`, throwing neither `schema 'doc' unknown` nor `schema 'foreign_schema' unknown` seem feasible.